src/node_http_common-inl.h

```diff
@@ -66,6 +68,341 @@ NgHeaders<T>::NgHeaders(Environment* env, v8::Local<v8::Array> headers) {
   }
 }
 
+static const size_t te_hash = std::hash<std::string>{}("te");
+static const size_t status_hash = std::hash<std::string>{}(":status");
+static const std::unordered_set<size_t> ng_forbidden_headers{
+  std::hash<std::string>{}("connection"),
+  std::hash<std::string>{}("upgrade"),
+  std::hash<std::string>{}("http2-settings"),
+  std::hash<std::string>{}("keep-alive"),
+  std::hash<std::string>{}("proxy-connection"),
+  std::hash<std::string>{}("transfer-encoding"),
+};
+static const std::unordered_set<size_t> ng_valid_pseudo_headers{
+  std::hash<std::string>{}(":status"),
+  std::hash<std::string>{}(":method"),
+  std::hash<std::string>{}(":authority"),
+  std::hash<std::string>{}(":scheme"),
+  std::hash<std::string>{}(":path"),
+  std::hash<std::string>{}(":protocol")
+};
+static const size_t ng_valid_pseudo_headers_size = ng_valid_pseudo_headers.size();
+
+MUST_USE_RESULT inline bool VALIDATE_FOR_ILLEGAL_CONNECTION_SPECIFIC_HEADER(
+        v8::Isolate*& isolate,
+        const std::string_view& header,
```

Comment:
string_view does not own or copy.

```suggestion
        const std::string_view header,
```